cumulus/parachains/integration-tests/emulated/common/src/macros.rs

```diff
@@ -822,3 +822,69 @@ macro_rules! test_cross_chain_alias {
 		}
 	};
 }
+
+/// note: $asset needs to be prefunded outside this function
+#[macro_export]
+macro_rules! create_pool_with_native_on {
+	( $chain:ident, $asset:expr, $is_foreign:expr, $asset_owner:expr ) => {
+		emulated_integration_tests_common::impls::paste::paste! {
+			<$chain>::execute_with(|| {
+				type RuntimeEvent = <$chain as Chain>::RuntimeEvent;
+				let owner = $asset_owner;
+				let signed_owner = <$chain as Chain>::RuntimeOrigin::signed(owner.clone());
+				let native_asset: Location = Parent.into();
+
+				if $is_foreign {
+					assert_ok!(<$chain as [<$chain Pallet>]>::ForeignAssets::mint(
+						signed_owner.clone(),
+						$asset.clone().into(),
+						owner.clone().into(),
+						10_000_000_000_000, // For it to have more than enough.
+					));
+				} else {
+					let asset_id = match $asset.interior.last() {
+						Some(GeneralIndex(id)) => *id as u32,
+						_ => unreachable!(),
+					};
+					assert_ok!(<$chain as [<$chain Pallet>]>::Assets::mint(
+						signed_owner.clone(),
+						asset_id.into(),
+						owner.clone().into(),
+						10_000_000_000_000, // For it to have more than enough.
+					));
+				}
+
+				assert_ok!(<$chain as [<$chain Pallet>]>::AssetConversion::create_pool(
+					signed_owner.clone(),
+					Box::new(native_asset.clone()),
+					Box::new($asset.clone()),
+				));
+
+				assert_expected_events!(
+					$chain,
+					vec![
+						RuntimeEvent::AssetConversion(pallet_asset_conversion::Event::PoolCreated { .. }) => {},
+					]
+				);
+
+				assert_ok!(<$chain as [<$chain Pallet>]>::AssetConversion::add_liquidity(
+					signed_owner,
+					Box::new(native_asset),
```

Comment:
@claravanstaden I am afraid of `native_asset`'s import `Location` - it works for polkadot-sdk where we use xcm::v5 for pallet_assets/pallet_asset_conversion, but will it work if those pallets are using older version (e.g. polkadot-fellows) and also the same for `GeneralIndex`? Would `Box::new(native_asset.into())` do the trick?

Reply:
@bkontur it doesn't seem to work. There are impls from v4 into v5, but not the other way around (from what I can see): https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/src/v5/location.rs#L748. I can change the method signature to pass in the `native_asset`? Not sure what to do about `GeneralIndex` though.